bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -75,6 +75,21 @@ def run_plan(scope, plan_name, named_args = {})
       )
     end
 
+    # Check for `$BOLTDIR/.resource_types` and create reference based on file name
+    boltdir_path = Puppet.lookup(:bolt_inventory).config.boltdir.path
+    static_loader = loaders.static_loader
+    static_loader.runtime_3_init
+    resource_dir = File.join(boltdir_path, '.resource_types')
+
+    if File.directory?(resource_dir)
+      Dir.entries(resource_dir).each do |resource_pp|
+        unless resource_pp.start_with?('.')
+          type_name = resource_pp[0..-4].capitalize
+          loaders.static_loader.send(:create_resource_type_reference, type_name)
```

Comment:
I'm not sure whether it's better to forcibly use the private method here, which ensures we match the implementation in Puppet, or to copy the implementation, which uses the public API. I think that would be:

```rb
loaders.set_entry(TypedName.new(:type, type_name), Types::TypeFactory.resource(type_name))
```

My slight preference may be the latter, but I'm really on the fence.

Reply:
I think using the public API is better. updated. Thanks.